test/evp_kdf_test.c

```diff
@@ -1678,6 +1678,132 @@ static int test_kdf_ss_kmac(void)
     return ret;
 }
 
+static 
+OSSL_PARAM *construct_snmpkdf_params(char *digest, unsigned char *eid, size_t eid_len,
+                                      unsigned char *password, size_t password_len)
+
+{
+    OSSL_PARAM *p = OPENSSL_malloc(sizeof(OSSL_PARAM) * 4);
+    OSSL_PARAM *p_ret = p;
+
+    if (!TEST_ptr(p))
+        return 0;
+
+    *p++ = OSSL_PARAM_construct_utf8_string(
+        OSSL_KDF_PARAM_DIGEST, digest, 0);
+    *p++ = OSSL_PARAM_construct_octet_string(
+        OSSL_KDF_PARAM_SNMPKDF_EID, eid, eid_len);
+    *p++ = OSSL_PARAM_construct_octet_string(
+        OSSL_KDF_PARAM_PASSWORD, password, password_len);
+    *p = OSSL_PARAM_construct_end();
+
+    return p_ret;
+}
+
+static int test_kdf_snmpkdf_null_output_buf(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+
+    static unsigned char EID1[] = {
+          0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+    unsigned char *password1 = NULL;
+
+    password1 = OPENSSL_malloc(1048577);
```

Comment:
Why the large buffer here.. If this is testing something else it should maybe point it out,,